source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/action/ActionElementRenderer.java

```diff
@@ -68,7 +74,54 @@ public ButtonOnClickListener(RenderedAdaptiveCard renderedCard, BaseActionElemen
         @Override
         public void onClick(View v)
         {
-            m_cardActionHandler.onAction(m_action, m_renderedAdaptiveCard);
+            if(m_action.GetElementType() == ActionType.ToggleVisibility)
+            {
+                ToggleVisibilityAction toggleVisibilityAction = null;
+                if (m_action instanceof ToggleVisibilityAction)
+                {
+                    toggleVisibilityAction = (ToggleVisibilityAction) m_action;
+                }
+                else if ((toggleVisibilityAction = ToggleVisibilityAction.dynamic_cast(m_action)) == null)
```

Comment:
Can you tell me why we need both of these checks? What is the case where both the if and the else if are false?

Reply:
As for the existence of the two checks, that's done for all the elements and actions when they are rendered, as far as I understand the first check is the Java casting version and the second check is the swig casting method; if both of those checks were to fail that means the object is not of the desired type and the action type was modified by something else but the action is not really a ToggleVisibility Action

---
In reply to: [251115166](https://github.com/Microsoft/AdaptiveCards/pull/2252#discussion_r251115166) [](ancestors = 251115166)